File: PA4.java
class MyHashSet {

    /** Initialize your data structure here. */
    
    boolean[][] arr ;
    public MyHashSet() {
       arr = new boolean[1000][];
    }
    public int getBucketId(int key){
        return key%1000;
    }
      public int getItemId(int key){
        return key/1001;
    }
    
    public void add(int key) {
        if(arr[getBucketId(key)]==null){
            arr[getBucketId(key)] = new boolean[1000];
        }
        arr[getBucketId(key)][getItemId(key)] = true;
    }
    
    public void remove(int key) {
        if(arr[getBucketId(key)] != null){
            arr[getBucketId(key)][getItemId(key)] = false;
        }
    }
    
    /** Returns true if this set contains the specified element */
    public boolean contains(int key) {
         if(arr[getBucketId(key)] != null){
           return arr[getBucketId(key)][getItemId(key)];
        }
        return false;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */

File: PA_4.py
#Leetcode 705 : Design HashSet
#Space Complexity - O(ParentBuckets * CHildBuckets) [Overall]
class MyHashSet(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.parentBuckets = 1000
        self.childBuckets = 1001
        self.hset = [None] * self.parentBuckets
    
    def getParentKey(self,key):
        #Time Complexity - O(1)
        return key % self.parentBuckets
    
    def getChildrenKey(self,key):
        #Time Complexity - O(1)
        return key // self.childBuckets

    def add(self, key):
        """
        :type key: int
        :rtype: None
        """
        #Time Complexity - O(1)
        parentKey = self.getParentKey(key)
        childKey = self.getChildrenKey(key)
        if self.hset[parentKey] == None:
            self.hset[parentKey] = [None] * self.childBuckets
        self.hset[parentKey][childKey] = True
        

    def remove(self, key):
        """
        :type key: int
        :rtype: None
        """
        #Time Complexity - O(1)
        parentKey = self.getParentKey(key)
        childKey = self.getChildrenKey(key)
        if self.hset[parentKey] != None:
            self.hset[parentKey][childKey] = None

    def contains(self, key):
        """
        Returns true if this set contains the specified element
        :type key: int
        :rtype: bool
        """
        #Time Complexity - O(1)
        parentKey = self.getParentKey(key)
        childKey = self.getChildrenKey(key)
        if self.hset[parentKey] != None:
            if self.hset[parentKey][childKey] != None:
                return True
        return False